### DISCUSSION TITLE
Azure Automation PowerShell 7.2

### DISCUSSION THREAD

**Discussion by jrudlin** [svrooij/WinTuner#62] [General]
How could we run this module in an Azure Automation Runbook (using Managed Identity)? The max version in a Runbook is 7.2 but the module requires 7.4
Or is there somewhere better to use the Managed Identity option?

**Comment by svrooij** [comment]
To give you some background, PowerShell 7.2 is built upon .NET6 (according to [this page](https://learn.microsoft.com/en-us/powershell/scripting/whats-new/differences-from-windows-powershell?view=powershell-7.4#net-framework-vs-net-core)). Since PowerShell 7.4 was released some time ago, and that is build upon .NET8 (current long-time-supported version), I thought I would upgrade this package to the latest LTS version of PowerShell (which is 7.4 at time of writting).

Apparently Microsoft does not support the latest LTS version of PowerShell on their own platform 😞.

![image](https://github.com/svrooij/WingetIntune/assets/1292510/0a58fbb0-74e2-431c-ae02-08b546bbe356)


Having said that, nothing really changed between the versions (for what I'm doing), so changing the minimum version back to 7.2 would be doable, will check this out next week.

Other ways are to use this module in either a GitHub action or an Azure DevOps pipeline. you have the ability to use DefaultAzureCredential there.

**Reply by jrudlin** [reply]
If there are no major differences in functionality with going back to 7.2 then I think it would be very useful (until Microsoft support 7.4 in Runbooks) for many people as this great module is likely to be used in that scenario, thanks.
